README.md

```diff
@@ -234,6 +234,14 @@ Message filters:
   - AllowSrcCompOut: If set, only allow messages from the listed MAVLink source
     compoent IDs to be sent via this endpoint
 
+Message de-duplication:
+
+  - If enabled, each incoming message is checked, whether another copy was
+    already received the last `DeduplicationPeriod` milliseconds ago. If it's
+    already known, the message will be dropped as it was never received and the
+    timeout counter for that message will be reset. Messages are identified via
+    their `std::hash` value.
```

Comment:
Either here or in the config.sample we need a note about potentially harmful side effect: if it's a link with packet loss, the re-transmission from either the transport or by the flightstack/mavlink-component could cause the re-transmissions to be dropped.

One case I think this is particularly bad: log streaming over mavlink. Example with the Ardupilot log format:

```
    <message id="184" name="REMOTE_LOG_DATA_BLOCK">
      <description>Send a block of log data to remote location.</description>
      <field type="uint8_t" name="target_system">System ID.</field>
      <field type="uint8_t" name="target_component">Component ID.</field>
      <field type="uint32_t" name="seqno" enum="MAV_REMOTE_LOG_DATA_BLOCK_COMMANDS">Log data block sequence number.</field>
      <field type="uint8_t[200]" name="data">Log data block.</field>
    </message>
```

Suppose the log is being saved by another endpoint (example: mavproxy) and not mavlink-router itself.

1. Flight stack sends seqno 1, we route it to the endpoint of whatever is saving the logs. 
2. Message gets lost
3. Logger checks seqno and notice it never received seqno, but after some time it receives seqno 2. Logger requests seqno 1 to be resent
4. Flight stack sends seqno 1
5. That message with seqno 1 will be dropped since it's exactly the same as the one previously sent

Logger here was an example, but there are probably several commands from the mavlink spec that would suffer from a similar issue. I'm not blocking the feature, but we need to document that this is known, and the user probably needs to tune the DeDuplicationPeriod for their application

Reply:
I'm not 100% sure, but wouldn't the re-transmitted packet have a different [packet sequence number](http://mavlink.io/en/guide/serialization.html#mavlink2_packet_format) in the MAVLink header? Since the hash is calculated over the full mavlink packet, this should give a different hash on the re-transmitted packet.

Reply:
Humn... right. I think I was misled by the `seqno` in this specific message - it needs to be the same in the re-transmitted message. However that is part of the payload, not the header, and the packet sequence number should take care of making the hash different as you noticed.

However, there is still the fact that packet sequence number is a uint8 only, so only covers 256 messages before rotating, which may be a very short time: that could cause occasional issues. However this is very less likely than the original issue I was worried about. Thanks.

Reply:
We actually came across the issue with the 8 bit header sequence number during the initial tests of de-duplication. Julian's first draft didn't remove the hashes from the cache, so heartbeats got dropped after some time. A grounded vehicle has a quite static heartbeat message and after some time all sequence numbers in the header were seen. IIRC this took a few minutes tough. But I'll try to test out a recommended max value.

Reply:
during those few minutes you experienced a steadily increasing "packet loss". I guess the safe configuration is to avoid the situation the sequence number could wrap and make the hashes collide... Take the stream with the highest rate that could be problematic due to its content, and define the timeout as less than 1/rate.